CMakeLists.txt

```diff
@@ -5,6 +5,24 @@ project(jsontoolkit VERSION 2.0.0 LANGUAGES CXX
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 include(vendor/noa/cmake/noa.cmake)
 
+# Enable IPO/LTO to help the compiler optimize across modules.
+# Only do so in release, given these optimizations can significantly
+# increase build times.
+# See: https://cmake.org/cmake/help/latest/module/CheckIPOSupported.html
+# TODO: Upstream to Noa
+if(CMAKE_BUILD_TYPE STREQUAL "Release" AND NOT BUILD_SHARED_LIBS)
+  include(CheckIPOSupported)
+  check_ipo_supported(RESULT ipo_supported OUTPUT ipo_supported_error)
+  if(ipo_supported)
+    message(STATUS "Enabling IPO")
+    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
```

Comment:
This means we enable it for everything from this point on-wards? Maybe better to enable it per target on `noa_library`?